js/glotdict-settings.js

```diff
@@ -18,21 +18,28 @@ if ( ( 'undefined' !== typeof $gp_editor_options ) && '1' === $gp_editor_options
 }
 gd_user.is_connected = document.querySelector( 'body.logged-in' ) !== null;
 
-jQuery( '#menu-headline-nav' ).append( '<li class="gd_setting" style="cursor:pointer;"><a> GlotDict</a></li>' );
-jQuery( '.gd_icon' ).prependTo( '.gd_setting' ).show();
+document.querySelector( '#menu-headline-nav' ).insertAdjacentHTML('beforeend', '<li class="gd_setting" style="cursor:pointer;"><a> GlotDict</a></li>');
+document.querySelector('.gd_setting').prepend(document.querySelector('.gd_icon'));
+document.querySelector('.gd_icon').style.display = "";
 
 const gd_settings_menu = document.querySelector( '.gd_setting' );
 gd_settings_menu && gd_settings_menu.addEventListener( 'click', () => {
+	const checkbox_listeners_flag = document.querySelector('.gp-content .gd_settings input[type="checkbox"]' ) ? true : false;
 	if ( document.body.classList.contains( 'gd-settings-on-screen' ) && '0' !== gd_extension.previousVersion ) {
 		gd_extension.previousVersion = gd_extension.currentVersion;
 		localStorage.setItem( 'gd_extension_status', JSON.stringify( gd_extension ) );
 	}
 	document.body.classList.toggle( 'gd-settings-on-screen' );
 	gd_generate_settings_panel();
-} );
 
-jQuery( '.gp-content' ).on( 'click', '.gd_settings input[type="checkbox"]', function() {
```

Comment:
I don't see a click event on this just a check of the checkbox status but is wrong we want that the value is changed on click not on the file loading.

Reply:
In both the current code and the PR, the click event handler is configured on the checkbox `.gp-content .gd_settings input[type="checkbox"]`. And when the checkbox is clicked we save the details and the state into the Local Storage (seen in line 35).

With Vanilla JS, the click event can only be configured after the checkboxes are rendered. This is the reason I have moved the checkbox code inside the `gd_settings_menu click function`. I have tested the checkbox functionality in different scenarios (opening and closing the GlotDict Settings), the code is working as required in all scenarios.

Reply:
Yes you are right I missed that line.
we need some testings :-D